src/components/screens/SendScreen/helpers/index.tsx

```diff
@@ -0,0 +1,130 @@
+import Blockaid from "@blockaid/client";
+import useAppTranslation from "hooks/useAppTranslation";
+import { useMemo } from "react";
+import { SecurityLevel } from "services/blockaid/constants";
+import {
+  assessTransactionSecurity,
+  extractSecurityWarnings,
+} from "services/blockaid/helper";
+
+/**
+ * Hook for managing transaction security assessment
+ * Extracts security warnings and determines severity level
+ */
+export function useTransactionSecurity(
+  transactionScanResult: Blockaid.StellarTransactionScanResponse | undefined,
+) {
+  const transactionSecurityAssessment = useMemo(
+    () => assessTransactionSecurity(transactionScanResult),
+    [transactionScanResult],
+  );
+
+  const transactionSecurityWarnings = useMemo(() => {
+    if (
+      transactionSecurityAssessment.isMalicious ||
+      transactionSecurityAssessment.isSuspicious
+    ) {
+      const warnings = extractSecurityWarnings(transactionScanResult);
+
+      if (Array.isArray(warnings) && warnings.length > 0) {
```

Comment:
Looks like `extractSecurityWarnings` is typed to return an Array. Should this type be updated if it can return anything other than an Array?

Reply:
If it's guarding against a type error or something similar, I think the helper should handle that so the caller has a reliable output